### ISSUE TITLE
📎 Support `--reporter=concise` for less verbose diagnostics

### ISSUE THREAD

**Issue by sterliakov** [biomejs/biome#6970]
### Description

Concise reporter is a way to report the diagnostics using less vertical space (and providing fewer details). 

This reporter can be enabled by requesting `--reporter=concise` in CLI args.

It works as follows (stderr and stdout use matches how they are now used by pretty formatter):

* All formatter diagnostics ("formatter would have printed the following content") are aggregated and reported to stderr in a trailing section titled "The following file(s) would be reformatted:" followed by a newline-separated indented list of violating files. 
* All other diagnostics are reported together to stderr, one line per diagnostic, containing:
  - severity icon
  - file path and lineno
  - full error code
  - short summary (one used as a "header" in current verbose formatter)
* The output ends with a summary written to stdout, reporting amount of checked files, warnings, errors, and files to be reformatted. 

Here's an example (colors should be similar to colors used by pretty reporter):

```
$ pbiome check a.ts b.ts
⚠ a.ts:1: lint/style/useImportType: Some named imports are only used as types.
⚠ a.ts:1: lint/correctness/noUnusedImports: These imports are unused.
✖ a.ts:3: lint/correctness/noUnusedVariables: This type alias Q is unused.
✖ a.ts:13: lint/correctness/noUnusedVariables: This function func1 is unused.
✖ a.ts:15: lint/correctness/noUnusedVariables: This function func2 is unused.
✖ b.ts:10: lint/correctness/noUnusedVariables: This function something is unused.
✖ b.ts:11: lint/correctness/noConstantCondition: Unexpected constant condition.

The following files would be reformatted:
    a.ts
    b.ts

Checked 2 files in 227ms. No fixes applied.
Found 5 errors.
Found 2 warnings.
Found 2 files that would be reformatted.
```

Yes, this means some rule summaries might be somewhat unclear. That can be improved  gradually and should not be a blocker for making concise formatter available.

Discussed&approved in #6498 and previously requested in #1909, see those for motivation and context.

**Comment by ematipico** [comment]
I believe you linked the wrong issue

**Comment by sterliakov** [comment]
Whoops, thanks!

**Comment by BenjaminBrienen** [comment]
Does this block the release of v2.2?

**Comment by sterliakov** [comment]
Sorry, I've had more load last week than I anticipated. I have a working implementation and need to finish tests for it; I hope to get it ready for review around Wednesday

**Comment by Andriamanitra** [comment]
For integration with editors and command line tools it would be ideal if the line-oriented format was similar to other linters, which is most commonly `filename:line:column: severity: message` (where `severity` is typically "note", "warning", or "error", but some linters also include an error id). This format is used by (among others):
* `quick-lint-js`
* `mypy`
* `cargo clippy --output-format=short`
* `gcc` [^1]
* `shellcheck --format=gcc`
* `clang` [^1]
* `rubocop --format=clang` [^1]

To motivate the use case of machine-readable concise output:
* JSON output (currently experimental) is not ideal because you need to install a tool like `jq` to parse it on the command-line, and editors plugins don't necessarily have access to a JSON parser either (Lua notably does not have one built-in).
* [nvim-lint](https://github.com/mfussenegger/nvim-lint/blob/master/lua/lint/linters/biomejs.lua) currently parses output of the default reporter, which is not really meant to be machine-readable.
* [micro's built-in linter.lua plugin](https://github.com/zyedidia/micro/blob/master/runtime/plugins/linter/linter.lua#L68-L92) expects a line-oriented format.

@ematipico suggested using an icon for internal consistency in https://github.com/biomejs/biome/discussions/6948#discussioncomment-13824178 which makes parsing slightly more annoying for (in my opinion) no real benefit, but it would not be too hard to work around. The missing column number is bit more of a bummer. Could these decisions be reconsidered?


[^1]: the output from gcc/clang is not concise, it also contains lines with leading whitespace that need to be filtered out

**Comment by ematipico** [comment]
I was under the impression that the proposal was a reporter for humans, not for machines. That's why I asked to keep the icon, for accessibility reasons.

But if @sterliakov proposal is meant to create a reporter for machines, then I suppose it's fine to remove the icon
